File: bindings/go/binding_test.go
package tree_sitter_xml_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_xml "github.com/tree-sitter-grammars/tree-sitter-xml/bindings/go"
)

func TestCanLoadXMLGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_xml.LanguageXML())
	if language == nil {
		t.Errorf("Error loading XML grammar")
	}
}

func TestCanLoadDTDGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_xml.LanguageDTD())
	if language == nil {
		t.Errorf("Error loading DTD grammar")
	}
}

File: bindings/node/binding_test.js
const assert = require("node:assert");
const { test } = require("node:test");

const Parser = require("tree-sitter");

test("can load XML grammar", () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require("./xml")));
});

test("can load DTD grammar", () => {
  const parser = new Parser();
  assert.doesNotThrow(() => parser.setLanguage(require("./dtd")));
});
